src/inc/engine/engine.h

```diff
@@ -13,22 +15,28 @@ class Engine
 {
 public:
 
+  Engine( std::vector<std::unique_ptr<GameComponent>>& components );
+
   void initialize( int height, int width );
 
-  void advance( std::vector<std::unique_ptr<GameComponent>>& components );
+  void advance();
```

Comment:
added a reference to the vector of game components as an instance of the engine. This couples an engine to a game, but makes the interface between the two cleaner.